// kuai/kui/ui/image.js
const GuaObject = require('../gua_object')
const ref = require('ref')
const Path = require('path')

const sdl = require('../../sdl')
const img = require('../../image')

const log = require('../utils').log

class ImageView extends GuaObject {
    constructor(position, path) {
        super()
        let p = Path.resolve(__dirname, '../../../', path)
        this.setup(position, p)
    }
    setup(position, path) {
        this.path = path
        // 添加 position 用来实现 background down/normal
        this.position = position
    }
    __added(view) {
        super.__added(view)
        let parent = this.parent
        this.renderer = parent.game.renderer
    }
    draw() {
        let imageFile = this.path
        this.surface = img.IMG_Load(imageFile)
        this.texture = sdl.SDL_CreateTextureFromSurface(this.renderer, this.surface)
        this.rect = new sdl.SDL_Rect({
            x: this.position[0],
            y: this.position[1],
            w: this.surface.deref().w,
            h: this.surface.deref().h,
        })
        sdl.SDL_RenderCopy(this.renderer, this.texture, null, this.rect.ref())
        sdl.SDL_FreeSurface(this.surface)
        sdl.SDL_DestroyTexture(this.texture)
    }
    remove() {
        sdl.SDL_DestroyTexture(this.texture)
    }
    
    setPosition(x, y) {
        const self = this
        self.position = [x, y]
    }
    getPosition(x, y) {
        const self = this
        return [...self.position]
    }
}


module.exports = ImageView
